### PR TITLE
"ConvertTests" rule improvements

### PR DESCRIPTION
**Pull request by FiveTimesTheFun**
This change updates UsesXunitForTestsFormattingRule ("ConvertTests")
to not remove unidentified usings (where the using cannot be mapped
to a symbol because the source of the using may not be in the
workspace), and to remove the ToF "ContractsRequiredAttribute".

### CODE REVIEW

File: src/Microsoft.DotNet.CodeFormatting/Rules/UsesXunitForTestsFormattingRule.cs [hunk 1]
```diff
@@ -52,6 +52,23 @@ internal UsesXunitForTestsFormattingRule(Options options)
 
             foreach (var usingSyntax in root.Usings)
             {
+                Action<UsingDirectiveSyntax> addUsing = (UsingDirectiveSyntax newUsingSyntax) =>
```
Comment (AlexGhiondea) [inline src/Microsoft.DotNet.CodeFormatting/Rules/UsesXunitForTestsFormattingRule.cs]: Should this be a separate method?

Comment (FiveTimesTheFun) [inline src/Microsoft.DotNet.CodeFormatting/Rules/UsesXunitForTestsFormattingRule.cs]: Sure.  I'll implement it as UsingDirectiveSyntax RemoveLeadingAndTrailingCompilerDirectives(this UsingDirectiveSyntax) and update the PR.

**Comment by jaredpar** [comment]
@FiveTimesTheFun can you sign the CLA for .NET contributions?

https://cla2.dotnetfoundation.org/

**Comment by ellismg** [comment]
He should have already signed it, for his work on CoreFx.

**Comment by jaredpar** [comment]
@ellismg if he has then that works fine.  The CLA bot got enabled after this PR was started so I just needed to manually verify the pre-existing PRs.

**Pull request merged**